src/twisted/application/runner/test/test_runner.py

```diff
@@ -68,11 +68,31 @@ def test_runInOrder(self):
         """
         L{Runner.run} calls the expected methods in order.
         """
-        runner = DummyRunner({})
+
+        calledMethods = []
```

Comment:
Superficially, it seems to me that the @durin42 & @nathanielmanistaatgoogle orthodoxy here would be to say that there's a clear interface here separable from `Runner` which is a different layer: the side-effect-y methods called out here (`killIfRequested`, `startLogging` etc).  However, this test appears to be the _only_ place in the entire codebase where we separate those layers, so perhaps it would be unnecessary overhead to actually separate them properly?

@ - mentions in place to see if the thought leaders in question have a thought to lead in this case.